src/lbaf/IO/lbsVTDataReader.py

```diff
@@ -84,9 +84,10 @@ def read_iteration(self, n_p: int, phase_id: int) -> list:
             # Try to retrieve rank information at given time-step
             try:
                 rank_list[p] = rank_iter_map[phase_id]
-            except KeyError:
-                self.__logger.error(f"Could not retrieve information for rank {p} at time_step {phase_id}")
-                sys.exit(1)
```

Comment:
Please restore the exit.

We are not using exceptions in this code.

Thanks.

Reply:
This is a sample unit test:
```Python3
    def test_lbs_vt_data_reader_read_iteration_key_error(self):
        with self.assertRaises(KeyError) as err:
            rank_list = self.lr.read_iteration(n_p=4, phase_id=5)
            print(err.exception.args[0])
        self.assertEqual(err.exception.args[0], "Could not retrieve information for rank 0 at time_step 5. KeyError 5")
```
If that is `sys.exit(1)` I CAN NOT TEST IT! Now I could prepare the test, so it goes into this `KeyError`, catch the message and check if that is the same a I am expecting.

Reply:
You cannot catch the return from a sys call?

Reply:
Point well taken, I am now convinced. 

Let's use exceptions and commit to never, ever, ever! try to catch them.
Thanks